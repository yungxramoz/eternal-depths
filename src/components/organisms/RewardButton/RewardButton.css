.reward-button {
  width: 100%;
}

.reward-button-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  column-gap: 1.5em;
  width: 100%;
  min-height: 80px;
}

.reward-icon {
  flex-grow: 0;
  flex-basis: 44px;
  flex-shrink: 0;
}

.reward-button-content {
  flex-grow: 1;
  text-align: left;
}

.reward-title {
  margin: 0;
}

.reward-subtitle {
  margin: 0 0 1em 0;
  font-size: x-small !important;
}

.reward-description {
  margin: 0;
  margin-top: 4px;
  font-size: xx-small !important;
  line-height: 18px !important;
}
