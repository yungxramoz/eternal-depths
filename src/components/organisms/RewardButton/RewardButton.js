import React from 'react'
import './RewardButton.css'
import GoldenButton from '../../molecules/GoldenButton/GoldenButton'
import RpgIcon from '../../atoms/RpgIcon/RpgIcon'

const RewardButton = ({
  onClick,
  disabled,
  icon,
  title,
  subtitle,
  description,
}) => {
  return (
    <GoldenButton
      className="reward-button"
      onClick={onClick}
      disabled={disabled}
    >
      <div className="reward-button-container">
        <RpgIcon icon={icon} />
        <div className="reward-button-content">
          <p className="reward-title">{title}</p>
          <p className="reward-subtitle">
            <i>{subtitle}</i>
          </p>
          <p className="reward-description">{description}</p>
        </div>
      </div>
    </GoldenButton>
  )
}

export default RewardButton
