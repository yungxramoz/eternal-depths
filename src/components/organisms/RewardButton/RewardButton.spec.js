import React from 'react'
import { render, fireEvent, screen } from '@testing-library/react'
import RewardButton from './RewardButton'

describe('RewardButton', () => {
  const props = {
    onClick: jest.fn(),
    disabled: false,
    icon: 'test-icon',
    title: 'Test Title',
    subtitle: 'Test Subtitle',
    description: 'Test Description',
  }

  it('renders correctly', () => {
    render(<RewardButton {...props} />)
    expect(document.body).toMatchSnapshot('RewardButton')
  })

  it('renders correctly when disabled', () => {
    render(<RewardButton {...props} disabled />)
    expect(document.body).toMatchSnapshot('RewardButton disabled')
  })

  it('calls onClick prop when clicked', () => {
    render(<RewardButton {...props} />)
    fireEvent.click(screen.getByText(props.title))
    expect(props.onClick).toHaveBeenCalledTimes(1)
  })
})
