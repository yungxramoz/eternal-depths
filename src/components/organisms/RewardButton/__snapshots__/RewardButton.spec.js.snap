// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardButton renders correctly when disabled: RewardButton disabled 1`] = `
<body>
  <div>
    <button
      class="golden-button reward-button"
      disabled=""
    >
      <div
        class="rpgui-container container framed framed-golden-2 golden-container"
      >
        <div
          class="reward-button-container"
        >
          <div
            class="rpgui-icon test-icon reward-icon"
          />
          <div
            class="reward-button-content"
          >
            <p
              class="reward-title"
            >
              Test Title
            </p>
            <p
              class="reward-subtitle"
            >
              <i>
                Test Subtitle
              </i>
            </p>
            <p
              class="reward-description"
            >
              Test Description
            </p>
          </div>
        </div>
      </div>
    </button>
  </div>
</body>
`;

exports[`RewardButton renders correctly: RewardButton 1`] = `
<body>
  <div>
    <button
      class="golden-button reward-button"
    >
      <div
        class="rpgui-container container framed framed-golden-2 golden-container"
      >
        <div
          class="reward-button-container"
        >
          <div
            class="rpgui-icon test-icon reward-icon"
          />
          <div
            class="reward-button-content"
          >
            <p
              class="reward-title"
            >
              Test Title
            </p>
            <p
              class="reward-subtitle"
            >
              <i>
                Test Subtitle
              </i>
            </p>
            <p
              class="reward-description"
            >
              Test Description
            </p>
          </div>
        </div>
      </div>
    </button>
  </div>
</body>
`;
